### PR TITLE
EventGrid .NET SDK: api-version 2020-10-15-preview

### PR DESCRIPTION
**Pull request by VidyaKukke**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [x] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [x] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running generate.ps1. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version,

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Comment by VidyaKukke** [comment]
Swagger for which sdk was generated: https://github.com/Azure/azure-rest-api-specs/tree/master/specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-10-15-preview

**Review by YalinLi0312 — COMMENTED** [review]
Could you update versions in *.csproj and AssemblyInfo.cs?

**Comment by VidyaKukke** [comment]
> Could you update versions in *.csproj and AssemblyInfo.cs?

It is updated?  It used to be 6.0.0 and now as part of this change it is 6.1.0. Is there something else you are looking for?

**Review by YalinLi0312 — APPROVED** [review]

File: sdk/eventgrid/Microsoft.Azure.Management.EventGrid/src/Generated/Models/EventSubscription.cs [hunk 1]
```diff
@@ -57,20 +65,35 @@ public EventSubscription()
...
+        /// Uses the managed identity setup on the parent resource (namely,
+        /// topic or domain) to acquire the authentication tokens being used
+        /// during delivery / dead-lettering.</param>
+        /// <param name="systemData">The system metadata relating to this
+        /// resource.</param>
+        public EventSubscription(string id = default(string), string name = default(string), string type = default(string), string topic = default(string), string provisioningState = default(string), EventSubscriptionDestination destination = default(EventSubscriptionDestination), DeliveryWithResourceIdentity deliveryWithResourceIdentity = default(DeliveryWithResourceIdentity), EventSubscriptionFilter filter = default(EventSubscriptionFilter), IList<string> labels = default(IList<string>), System.DateTime? expirationTimeUtc = default(System.DateTime?), string eventDeliverySchema = default(string), RetryPolicy retryPolicy = default(RetryPolicy), DeadLetterDestination deadLetterDestination = default(DeadLetterDestination), DeadLetterWithResourceIdentity deadLetterWithResourceIdentity = default(DeadLetterWithResourceIdentity), SystemData systemData = default(SystemData))
```
Comment (bquantump) [inline sdk/eventgrid/Microsoft.Azure.Management.EventGrid/src/Generated/Models/EventSubscription.cs:82]: Breaking change

File: sdk/eventgrid/Microsoft.Azure.Management.EventGrid/src/Generated/Models/EventSubscriptionFilter.cs [hunk 2]
```diff
@@ -45,14 +45,18 @@ public EventSubscriptionFilter()
...
         /// SubjectBeginsWith and SubjectEndsWith properties of the filter
         /// should be compared in a case sensitive manner.</param>
+        /// <param name="enableAdvancedFilteringOnArrays">Allows advanced
+        /// filters to be evaluated against an array of values instead of
+        /// expecting a singular value.</param>
         /// <param name="advancedFilters">An array of advanced filters that are
         /// used for filtering event subscriptions.</param>
-        public EventSubscriptionFilter(string subjectBeginsWith = default(string), string subjectEndsWith = default(string), IList<string> includedEventTypes = default(IList<string>), bool? isSubjectCaseSensitive = default(bool?), IList<AdvancedFilter> advancedFilters = default(IList<AdvancedFilter>))
+        public EventSubscriptionFilter(string subjectBeginsWith = default(string), string subjectEndsWith = default(string), IList<string> includedEventTypes = default(IList<string>), bool? isSubjectCaseSensitive = default(bool?), bool? enableAdvancedFilteringOnArrays = default(bool?), IList<AdvancedFilter> advancedFilters = default(IList<AdvancedFilter>))
```
Comment (bquantump) [inline sdk/eventgrid/Microsoft.Azure.Management.EventGrid/src/Generated/Models/EventSubscriptionFilter.cs:53]: breaking change

File: sdk/eventgrid/Microsoft.Azure.Management.EventGrid/src/Generated/Models/EventSubscriptionUpdateParameters.cs [hunk 3]
```diff
@@ -46,17 +54,29 @@ public EventSubscriptionUpdateParameters()
...
+        /// <param name="deadLetterWithResourceIdentity">The dead letter
+        /// destination of the event subscription. Any event that cannot be
+        /// delivered to its' destination is sent to the dead letter
+        /// destination.
+        /// Uses the managed identity setup on the parent resource (topic /
+        /// domain) to acquire the authentication tokens being used during
+        /// delivery / dead-lettering.</param>
+        public EventSubscriptionUpdateParameters(EventSubscriptionDestination destination = default(EventSubscriptionDestination), DeliveryWithResourceIdentity deliveryWithResourceIdentity = default(DeliveryWithResourceIdentity), EventSubscriptionFilter filter = default(EventSubscriptionFilter), IList<string> labels = default(IList<string>), System.DateTime? expirationTimeUtc = default(System.DateTime?), string eventDeliverySchema = default(string), RetryPolicy retryPolicy = default(RetryPolicy), DeadLetterDestination deadLetterDestination = default(DeadLetterDestination), DeadLetterWithResourceIdentity deadLetterWithResourceIdentity = default(DeadLetterWithResourceIdentity))
```
Comment (bquantump) [inline sdk/eventgrid/Microsoft.Azure.Management.EventGrid/src/Generated/Models/EventSubscriptionUpdateParameters.cs:69]: breaking change

File: sdk/eventgrid/Microsoft.Azure.Management.EventGrid/src/Generated/Models/TopicUpdateParameters.cs [hunk 4]
```diff
@@ -43,11 +44,14 @@ public TopicUpdateParameters()
         /// <param name="inboundIpRules">This can be used to restrict traffic
         /// from specific IPs instead of all IPs. Note: These are considered
         /// only if PublicNetworkAccess is enabled.</param>
-        public TopicUpdateParameters(IDictionary<string, string> tags = default(IDictionary<string, string>), string publicNetworkAccess = default(string), IList<InboundIpRule> inboundIpRules = default(IList<InboundIpRule>))
+        /// <param name="sku">The Sku pricing tier for the topic.</param>
+        public TopicUpdateParameters(IDictionary<string, string> tags = default(IDictionary<string, string>), IdentityInfo identity = default(IdentityInfo), string publicNetworkAccess = default(string), IList<InboundIpRule> inboundIpRules = default(IList<InboundIpRule>), ResourceSku sku = default(ResourceSku))
         {
```
Comment (bquantump) [inline sdk/eventgrid/Microsoft.Azure.Management.EventGrid/src/Generated/Models/TopicUpdateParameters.cs:49]: Breaking change

File: sdk/eventgrid/Microsoft.Azure.Management.EventGrid/src/Generated/Models/EventSubscription.cs (hunk 1 above)
Comment (bquantump) [reply sdk/eventgrid/Microsoft.Azure.Management.EventGrid/src/Generated/Models/EventSubscription.cs:82]: Going to pump the API versions to "fix" this

File: sdk/eventgrid/Microsoft.Azure.Management.EventGrid/src/Generated/Models/EventSubscriptionUpdateParameters.cs (hunk 3 above)
Comment (bquantump) [reply sdk/eventgrid/Microsoft.Azure.Management.EventGrid/src/Generated/Models/EventSubscriptionUpdateParameters.cs:69]: Going to pump the API versions to "fix" this

File: sdk/eventgrid/Microsoft.Azure.Management.EventGrid/src/Generated/Models/TopicUpdateParameters.cs (hunk 4 above)
Comment (bquantump) [reply sdk/eventgrid/Microsoft.Azure.Management.EventGrid/src/Generated/Models/TopicUpdateParameters.cs:49]: Going to pump the API versions to "fix" this

File: sdk/eventgrid/Microsoft.Azure.Management.EventGrid/src/Generated/Models/EventSubscriptionFilter.cs (hunk 2 above)
Comment (bquantump) [reply sdk/eventgrid/Microsoft.Azure.Management.EventGrid/src/Generated/Models/EventSubscriptionFilter.cs:53]: Going to pump the API versions to "fix" this

**Pull request merged**